feat: audit scanner docs. (#199)
kubewarden/docs: docs/howtos/audit-scanner.md

---
sidebar_label: "Audit Scanner Installation"

> title: "Kubewarden Audit Scanner Installation"

Reviewer: ```suggestion
title: "Audit Scanner Installation"
```

---

> # Kubewarden Audit Scanner installation

Reviewer: ```suggestion
# Audit Scanner installation
```

Starting from version `v1.7.0`, Kubewarden introduces a new component called
the Audit Scanner. This scanner evaluates whether the resources running in the
cluster comply with the deployed policies. It is particularly useful when
policies are deployed after the resources. The Audit Scanner allows operators
to detect resources deployed before the policies that violate the validations
performed by the policies.

## Installation

> To install and use the Kubewarden Audit Scanner, please follow these steps:

Reviewer: ```suggestion
```

I would remove this line and go straight to the steps. Less to read, same info.

1. Install the `kubewarden-controller` Helm chart, ensuring you have version
   `v1.7.0` or higher. Remember to enable the audit scanner component.

   ```console

>    helm install kubewarden-crds kubewarden/kubewarden-crds

Reviewer: The step says install `kubewarden-controller`, but here we are installing the crds. I suppose we want a step 1 that installs the crds, were we say that they by default come with the PolicyReports CRDs, and we have the `::caution` paragraph. And then a step 2 for installing the kubewarden controller, enabling the audit scanner, and talking about its defaults.

   helm install --set auditScanner.enable=true kubewarden-controller kubewarden/kubewarden-controller
   ```

   For more information about the installation of Kubewarden see the [Quick Start guide](../quick-start.md)

:::caution
   The PolicyReport Custom Resource Definitions must be available to store the
   policy report results. Therefore, if the `kubewarden-crds` is installed with
   the `installPolicyReportCRDs` value set to `false`, the cluster operator should
   install the CRD manually. Note that, the `kubewarden-crds` installs the CRDs
   by default.

   See more info about the CRDs at the [policy work group
   repository](https://github.com/kubernetes-sigs/wg-policy-prototypes)
:::

By default, the Audit Scanner is implemented as a cron job that will be
triggered every 60 minutes. You can adjust this and other audit scanner

> configuration changing other chart values. Check the

Reviewer: ```suggestion
configuration changing the kubewarden-controller chart values. Check the
```

> [values.yaml](https://github.com/kubewarden/helm-charts/blob/main/charts/kubewarden-controller/values.yaml)

Reviewer: I would remove the line "Check the file for more options" as it is understood from the previous line.

file for more options.  Please note that the successful installation and
functioning of the Audit Scanner depend on the proper installation of the
Policy Report CRDs. Without the CRDs, the scanner will not be able to store and
retrieve the audit results accurately.

> See [here](../explanations/audit-scanner) more information about the Audit Scanner

Reviewer: ```suggestion
See [here](../explanations/audit-scanner) more information about the Audit Scanner.
```

---
sidebar_label: "Audit Scanner Installation"
title: "Kubewarden Audit Scanner Installation"
---

# Kubewarden Audit Scanner installation

Starting from version `v1.7.0`, Kubewarden introduces a new component called
the Audit Scanner. This scanner evaluates whether the resources running in the
cluster comply with the deployed policies. It is particularly useful when
policies are deployed after the resources. The Audit Scanner allows operators
to detect resources deployed before the policies that violate the validations
performed by the policies.

## Installation

To install and use the Kubewarden Audit Scanner, please follow these steps:

1. Install the `kubewarden-controller` Helm chart, ensuring you have version
   `v1.7.0` or higher. Remember to enable the audit scanner component.

   ```console
   helm install --set experimental.auditScanner.enable=true kubewarden-controller kubewarden/kubewarden-controller
   ```

   For more information about the installation of Kubewarden see the [Quick Start guide](../quick-start.md)

> 2. Install the Policy Report CRDs, which are required for the Audit Scanner to

Reviewer: this isn't needed, they are included by default on `kubewarden-crds`, unless one sets  `installPolicyReportCRDs` to `false`.

---
sidebar_label: "Audit Scanner Installation"
title: "Audit Scanner Installation"
---

# Audit Scanner installation

Starting from version `v1.7.0`, Kubewarden introduces a new component called
the Audit Scanner. This scanner evaluates whether the resources running in the
cluster comply with the deployed policies. It is particularly useful when
policies are deployed after the resources. The Audit Scanner allows operators
to detect resources deployed before the policies that violate the validations
performed by the policies.

## Installation

1. Install the `kubewarden-crds` Helm chart, ensuring you have app version

>    `v1.7.0` or higher. Remember to enable the audit scanner component.

Reviewer: ```suggestion
   `v1.7.0` or higher. The chart installs the PolicyReports CRDs by default.
```

   ```console
   helm install kubewarden-crds kubewarden/kubewarden-crds
   ```
   :::caution

>       The PolicyReport Custom Resource Definitions must be available to store the

Reviewer: This reads like an overexplanation: "if you don't install them then they don't get installed".
Maybe we can change it to provide reasons. e.g:

> Caution: in case that the needed PolicyReport CRDs are already present in the cluster, one will not be able to install them via the `kubewarden-crds` chart. One can then set `installPolicyReportCRDs` to `false` for the chart, and then the Kubewarden stack will not own those CRDs, but the admin.

      policy report results. Therefore, if the `kubewarden-crds` is installed with
      the `installPolicyReportCRDs` value set to `false`, the cluster operator should
      install the CRD manually. Note that, the `kubewarden-crds` installs the CRDs
      by default.

      See more info about the CRDs at the [policy work group
      repository](https://github.com/kubernetes-sigs/wg-policy-prototypes)
   :::

2. Install the `kubewarden-controller` Helm chart, ensuring you have app version
   `v1.7.0` or higher. Remember to enable the audit scanner component.

   ```console
   helm install --set auditScanner.enable=true kubewarden-controller kubewarden/kubewarden-controller
   ```

   :::note
   The audit scanner is not enabled by default. The operator should enabled it using
   Helm chart values.
   :::

   For more information about the installation of Kubewarden see the [Quick Start guide](../quick-start.md)

> By default, the Audit Scanner is implemented as a cron job that will be

Reviewer: ```suggestion
By default, the Audit Scanner is implemented as a [Cronjob](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/ that will be
```

triggered every 60 minutes. You can adjust this and other audit scanner
configuration changing the kubewarden-controller chart
[values.yaml](https://github.com/kubewarden/helm-charts/blob/main/charts/kubewarden-controller/values.yaml).

> Please note that the successful installation and functioning of the Audit

Reviewer: This reminder feels repeated from the one in point 1.

---
sidebar_label: "Audit Scanner Installation"
title: "Audit Scanner Installation"
---

# Audit Scanner installation

Beginning with version `v1.7.0`, Kubewarden has a new feature called "Audit Scanner".
A new component, called "audit-scanner", constantly checks the resources declared in the

> cluster, flagging the ones that do not adhere with the deployed Kubewarden policies.
> Policies evolve over the time: new ones are deployed and the existing ones can be

Reviewer 2: Splitting into 2 paragraphs:

```suggestion
cluster, flagging the ones that do not adhere with the deployed Kubewarden policies.

Policies evolve over the time: new ones are deployed and the existing ones can be
```

updated, both in terms of version and configuration settings.
This can lead to situations where resources already inside of the cluster

> are no longer compliant. The audit scanner feature provides Kubernetes administrators
> with a tool to consistently verify the compliance state of their clusters.

Reviewer 2: same here:

```suggestion
are no longer compliant.

The audit scanner feature provides Kubernetes administrators
with a tool to consistently verify the compliance state of their clusters.
```

## Installation

The audit scanner component is available since Kubewarden `v1.7.0`. Therefore,
make sure you are installing the Helm chart with app version `v1.7.0` or
higher.

1. Install the `kubewarden-crds` Helm chart. The chart install the needed
   `PolicyReport` CRDs by default.

   ```console
   helm install kubewarden-crds kubewarden/kubewarden-crds
   ```
   :::caution
      To store the results of policy reports, you need to have the PolicyReport
      Custom Resource Definitions (CRDs) available. If the necessary
      PolicyReport CRDs are already in the cluster, you cannot install them
      using the kubewarden-crds chart. In such case, you can disable the
      installation of PolicyReport CRDs by setting `installPolicyReportCRDs` to
      `false` in the chart. This means that the Kubewarden stack will not manage
      those CRDs, and the responsibility will be with the administrator.

      See more info about the CRDs at the [policy work group
      repository](https://github.com/kubernetes-sigs/wg-policy-prototypes)
   :::

2. Install the `kubewarden-controller` Helm chart. Remember to enable the audit
   scanner component.

   ```console

>    helm install --set auditScanner.enable=true kubewarden-controller kubewarden/kubewarden-controller

Reviewer 2: Not related with the docs, but shouldn't we enable the audit scanner by default?

CC @viccuad, @jvanz

Author: I'm fine with that as well.

Author: https://github.com/kubewarden/helm-charts/pull/262

Author: As audit scanner will be enable by default. Does it make sense to have this doc page?

   ```

   :::note
   The audit scanner is not enabled by default. It must be enabled using
   Helm chart values.
   :::

   For more information about the installation of Kubewarden see the [Quick Start guide](../quick-start.md)

By default, the Audit Scanner is implemented as a
[Cronjob](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs)
that will be triggered every 60 minutes. You can adjust this and other audit

> scanner configuration changing the kubewarden-controller chart

Reviewer 2: ```suggestion
scanner settings by changing the kubewarden-controller chart
```

The audit scanner feature provides Kubernetes administrators
with a tool to consistently verify the compliance state of their clusters.

## Installation

The audit scanner component is available since Kubewarden `v1.7.0`. Therefore,
make sure you are installing the Helm chart with app version `v1.7.0` or
higher.

1. Install the `kubewarden-crds` Helm chart. The chart install the needed
   `PolicyReport` CRDs by default.

   ```console
   helm install kubewarden-crds kubewarden/kubewarden-crds
   ```
   :::caution
      To store the results of policy reports, you need to have the PolicyReport
      Custom Resource Definitions (CRDs) available. If the necessary
      PolicyReport CRDs are already in the cluster, you cannot install them
      using the kubewarden-crds chart. In such case, you can disable the
      installation of PolicyReport CRDs by setting `installPolicyReportCRDs` to
      `false` in the chart. This means that the Kubewarden stack will not manage
      those CRDs, and the responsibility will be with the administrator.

      See more info about the CRDs at the [policy work group
      repository](https://github.com/kubernetes-sigs/wg-policy-prototypes)
   :::

> 2. Install the `kubewarden-controller` Helm chart. Remember to enable the audit

Reviewer: Good catch, fixed it with https://github.com/kubewarden/docs/pull/199/commits/8f37ab6a07a6f2360d661571be94935d35f69fda.
